pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -1007,3 +1013,60 @@ message ImageFsInfoResponse {
     // filesystem information of images.
     FsInfo fs_info = 1;
 }
+
+message ContainerStatsRequest{
+    // ID of the container for which to retrieve stats.
+    string container_id = 1;
+}
+
+message ContainerStatsResponse {
+    // Stats of the container.
+    ContainerStats stats = 1;
+}
+
+message ListContainerStatsRequest{
+}
+
+message ListContainerStatsResponse {
+    // Stats of the container.
+    repeated ContainerStats stats = 1;
+}
+
+// ContainerStats provides the resource usage statistics for a container.
+message ContainerStats {
+    // Information of the container.
+    Container container = 1;
+    // CPU usage gathered from the container.
+    CpuUsage cpu_usage = 2;
+    // Memory usage gathered from the container.
+    MemoryUsage memory_usage = 3;
+    // Root filesystem usage gathered from the container.
+    FilesystemUsage root_filesystem_usage = 4;
+}
+
+// CpuUsage provides the CPU usage information.
+message CpuUsage {
+    // Timestamp in nanoseconds at which the information were collected. Must be > 0.
+    int64 timestamp = 1;
+    // Cumulative CPU usage (sum of all cores) since object creation.
+    UInt64Value UsageCoreNanoSeocnds = 2;
+}
+
+// CpuUsage provides the memory usage information.
+message MemoryUsage {
+    // Timestamp in nanoseconds at which the information were collected. Must be > 0.
+    int64 timestamp = 1;
+    // The amount of working set memory in bytes.
+    UInt64Value WorkingSetBytes  = 2;
```

Comment:
WorkingSet isn't well defined outside of cAdvisor currently. At the CRI level can we instead get more detailed metrics - essentially the output of `memory.stat` from cgroups?